tests/integration_tests/utils.py

```diff
@@ -961,6 +965,54 @@ def assert_all_required_metrics_exist(
             ), f"required metrics {required_metric_names} not in metrics {metric_names} for feature {feature_name}"
 
 
+def assert_preprocessed_dataset_shape_and_dtype_for_feature(
+    feature_name: str,
+    preprocessed_dataset: "Dataset",
+    config_obj: "ModelConfig",
+    expected_dtype: np.dtype,
+    expected_shape: Tuple,
+):
+    """Asserts that the preprocessed dataset has the correct shape and dtype for a given feature type.
+
+    Args:
+        feature_name: the name of the feature to check
+        preprocessed_dataset: the preprocessed dataset
+        config_obj: the model config object
+        expected_dtype: the expected dtype
+        expected_shape: the expected shape
+    Returns:
+        None.
+    Raises:
+        AssertionError if the preprocessed dataset does not have the correct shape and dtype for the given feature type.
+    """
+    if_configs = [if_config for if_config in config_obj.input_features if if_config.name == feature_name]
+    # fail fast if given `feature_name`` is not found or is not unique
+    if len(if_configs) != 1:
+        raise ValueError(f"feature_name {feature_name} found {len(if_configs)} times in config_obj")
+    if_config = if_configs[0]
+
+    if_config_proc_column = if_config.proc_column
+    for result in [
+        preprocessed_dataset.training_set,
+        preprocessed_dataset.validation_set,
+        preprocessed_dataset.test_set,
+    ]:
+        result_df = result.to_df()
+        result_df_image_col = result_df[if_config_proc_column]
```

Comment:
Nit: it looks like this function is named generically for any feature type, but image is used here.

Reply:
Good catch! Fixed in latest.